app/assets/javascripts/discourse/app/components/user-nav.js

```diff
@@ -15,4 +20,27 @@ export default class UserNav extends Component {
       ? I18n.t("drafts.label_with_count", { count })
       : I18n.t("drafts.label");
   }
+
+  @bind
+  _handleClickEvent(event) {
+    if (!event.target.closest(`.${DROPDOWN_BUTTON_CSS_CLASS}`)) {
+      next(() => {
```

Comment:
what's the reason of the next here ?

Reply:
Because we’re clicking a link within the element, removing the element and hence the link before the transition has completed will actually cause the transition to become a full page reload.